deploy/charts/harvester/templates/configmap.yaml

```
@@ -93,21 +93,38 @@ data:
     - rke2-snapshot-validation-webhook
     EOF
 
+    # make sure we should not have any related label/taint on the node
```

this fails since kubectl returns a non zero respone when attempting to remote a taint when one is not present and promotion fails 
```
error: taint "node-role.kubernetes.io/etcd:NoExecute" not found 
```
This can be fixed by checking if taint exists before attempting to remove it.

```suggestion
      # make sure we should not have any related label/taint on the node
    if [[ $ETCD_ONLY == false ]]; then
      found=$($KUBECTL get node $HOSTNAME -o yaml | $YQ '.spec.taints[] | select (.effect == "NoSchedule" and .key == "node-role.kubernetes.io/etcd=true") | .effect')
      if [[ -n $found ]]
      then
        $KUBECTL taint nodes $HOSTNAME node-role.kubernetes.io/etcd=true:NoExecute-
      fi
      $KUBECTL label --overwrite nodes $HOSTNAME node-role.harvesterhci.io/witness-
    fi
```

---

Nice catch! Updated.
Please help to recheck.